### PR TITLE
fixed DataSource autocommit handling without UserTransaction

### PR DESCRIPTION
**Pull request by sergiorussia**
rewrote TxTransactionManagerTest to SqlClosureTest. added combination of datasource with and w/o autocommit / TxMan - and fixed one of combination (removed hasTxManager check as i mentioned in #20)

### CODE REVIEW

File: src/test/java/com/zaxxer/sansorm/SqlClosureTest.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+   @Test
+   public void shouldRollbackNestedClosuresWithUserTransaction() {
+      assertThatThrownBy(() -> SqlClosure.sqlExecute(c -> {
+         SqlClosureElf.executeUpdate(c, "INSERT INTO tx_test VALUES (?)", "6");
+
+         // here goes nested SqlClosure
+         SqlClosure.sqlExecute(cNested -> SqlClosureElf.executeUpdate(cNested, "INSERT INTO tx_test VALUES (?)", "7"));
+
+         SqlClosureElf.executeUpdate(c, "INSERT INTO tx_test VALUES (?)", "8");
+         throw new Error("boom!"); // ie something not or type SQLException or RuntimeException
```
Comment (sergiorussia) [inline src/test/java/com/zaxxer/sansorm/SqlClosureTest.java]: throwing Error previously would not rollback tx

File: src/test/java/com/zaxxer/sansorm/SqlClosureTest.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@RunWith(Parameterized.class)
+public class SqlClosureTest {
+   @Parameterized.Parameters(name = "autocommit={0}, ut={1}")
+   public static Collection<Object[]> data() {
+      return Arrays.asList(new Object[][] {
+         { true, true }, { true, false }, { false, true }, { false, false }
```
Comment (sergiorussia) [inline src/test/java/com/zaxxer/sansorm/SqlClosureTest.java]: autocommit=true, ut=false failed previously

**Comment by sergiorussia** [comment]
@brettwooldridge i ask you to release a new build after this PR :)

**Comment by brettwooldridge** [comment]
I’m away from my PC for a couple of hours, but I’ll cut a release later tonight.

**Comment by brettwooldridge** [comment]
@sergiorussia v3.2 has been published.

**Pull request merged**